test/e2e/test/maps/checks.go

```diff
@@ -0,0 +1,150 @@
+					Labels: map[string]string{
+						"eck.k8s.elastic.co/credentials":            "true",
+						"elasticsearch.k8s.elastic.co/cluster-name": b.EMS.Spec.ElasticsearchRef.Name,
+						"mapsassociation.k8s.elastic.co/name":       emsName,
+						"mapsassociation.k8s.elastic.co/namespace":  b.EMS.Namespace,
+					},
+				},
+			)
+		}
+		if b.EMS.Spec.HTTP.TLS.Enabled() {
+			expected = append(expected,
+				test.ExpectedSecret{
+					Name: emsName + "-ems-http-ca-internal",
+					Keys: []string{"tls.crt", "tls.key"},
+					Labels: map[string]string{
+						"maps.k8s.elastic.co/name":   emsName,
+						"common.k8s.elastic.co/type": "maps",
+					},
+				},
+				test.ExpectedSecret{
+					Name: emsName + "-ems-http-certs-internal",
+					Keys: []string{"tls.crt", "tls.key", "ca.crt"},
+					Labels: map[string]string{
+						"maps.k8s.elastic.co/name":   emsName,
+						"common.k8s.elastic.co/type": "maps",
+					},
+				},
+				test.ExpectedSecret{
+					Name: emsName + "-ems-http-certs-public",
+					Keys: []string{"ca.crt", "tls.crt"},
+					Labels: map[string]string{
+						"maps.k8s.elastic.co/name":   emsName,
+						"common.k8s.elastic.co/type": "maps",
+					},
+				},
+			)
+		}
+		return expected
+	})
+}
+
+func CheckStatus(b Builder, k *test.K8sClient) test.Step {
+	return test.Step{
+		Name: "Elastic Maps Server status should be updated",
+		Test: test.Eventually(func() error {
+			var ems v1alpha1.ElasticMapsServer
+			if err := k.Client.Get(context.Background(), k8s.ExtractNamespacedName(&b.EMS), &ems); err != nil {
+				return err
+			}
+			// don't check the association status that may vary across tests
+			ems.Status.AssociationStatus = ""
+			expected := v1alpha1.MapsStatus{
+				DeploymentStatus: commonv1.DeploymentStatus{
+					AvailableNodes: b.EMS.Spec.Count,
+					Version:        b.EMS.Spec.Version,
+					Health:         "green",
+				},
+				AssociationStatus: "",
+			}
+			if ems.Status != expected {
+				return fmt.Errorf("expected status %+v but got %+v", expected, ems.Status)
+			}
+			return nil
+		}),
+	}
+}
+
+type emsStatus struct {
+	Version string `json:"version"`
+	Overall struct {
+		State string `json:"state"`
+	} `json:"overall"`
+}
+
+func (b Builder) CheckStackTestSteps(k *test.K8sClient) test.StepList {
+	println(test.Ctx().TestTimeout)
+	return test.StepList{
+		{
+			Name: "Elastic Maps Server should respond to requests",
+			Test: test.Eventually(func() error {
+				client, err := NewMapsClient(b.EMS, k)
+				if err != nil {
+					return err
+				}
+				bytes, err := DoRequest(client, b.EMS, "GET", "/status")
+				if err != nil {
+					return err
+				}
+				var status emsStatus
+				if err := json.Unmarshal(bytes, &status); err != nil {
+					return err
+				}
+				// we cannot just test for "available" because in absence of a valid license the state is "critical"
+				// however we consider a parseable response to be a sufficient signal for a successful deployment
```

Comment:
While running the tests I mostly got the `degraded` status because of the tiles which are not installed. I'm wondering if a test should fail for `unavailable` or `critical`, but as I don't know the real meaning of these status I think I'm fine with the current approach.

Reply:
Hm good point. I think you are right we could test for that. You are right in 7.11-7.12 it returns `degraded` if the tile database is missing. Starting with 7.13 it will be `available` even in the absence of the tiles, because maps server will start  to support running with smaller basemaps.